src/App.tsx

```diff
@@ -15,47 +18,109 @@ export const goodsFromServer = [
+  ALPABET,
+  LENGTH,
+}
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export function getSortedGoods(
+  goods: string[],
+  state: State,
+) {
+  const { sortType, isReversed } = state;
+  const visibleGoods = [...goods].sort((goodA, goodB) => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return goodA.localeCompare(goodB);
+
+      case SortType.LENGTH:
+        return goodA.length - goodB.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const resetOrder = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const visibleGoods = getSortedGoods(goodsFromServer,
+    { sortType, isReversed });
+
+  const shouldRenderReset = isReversed || sortType !== SortType.NONE;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
+        <Button
+          className={classNames(
+            'button',
+            'is-info',
+            { 'is-light': sortType !== SortType.ALPABET },
+          )}
+          onClick={() => setSortType(SortType.ALPABET)}
         >
           Sort alphabetically
-        </button>
+        </Button>
 
-        <button
-          type="button"
-          className="button is-success is-light"
+        <Button
+          className={classNames(
+            'button',
+            'is-success',
+            { 'is-light': sortType !== SortType.LENGTH },
+          )}
+          onClick={() => setSortType(SortType.LENGTH)}
         >
           Sort by length
-        </button>
+        </Button>
 
-        <button
-          type="button"
-          className="button is-warning is-light"
+        <Button
+          className={classNames(
+            'button',
+            'is-warning',
+            { 'is-light': !isReversed },
+          )}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
-        </button>
+        </Button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {shouldRenderReset && (
+          <Button
+            className="button is-danger is-light"
+            onClick={resetOrder}
+          >
+            Reset
+          </Button>
+        )}
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {visibleGoods.map(good => (
+            <li
+              data-cy="Good"
+              key={uuidv4()}
```

Comment:
you should not generate random key. That means you gonna rerender a list with each component rendering - you definitely don't want it to be that way.